modules/transactionpool/transactionpool_test.go

```diff
@@ -473,3 +473,195 @@ func TestTpoolScalability(t *testing.T) {
+		if err != nil {
+			t.Fatal(err)
+		}
+	}
+	block, err := tpt.miner.AddBlock()
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// Now accept all the other nodes of each graph.
+	for _, graph := range graphs {
+		for _, txn := range graph[1:] {
+			err := tpt.tpool.AcceptTransactionSet([]types.Transaction{txn})
+			if err != nil {
+				t.Fatal(err)
+			}
+		}
+	}
+	// Now we mine 2 blocks in sequence and check that higher fee transactions
+	// show up to the first block.
+	block, err = tpt.miner.AddBlock()
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	var totalFee1 types.Currency
+	expectedFee1 := coinFrac.Mul64(321915)
+
+	// Add up total fees
+	numTxns1 := 0
+	maxFee1 := types.SiacoinPrecision.Div64(1000000)
+	minFee1 := types.SiacoinPrecision.Mul64(1000000)
+	for _, txn := range block.Transactions {
+		for _, fee := range txn.MinerFees {
+			if fee.Cmp(maxFee1) >= 0 {
+				maxFee1 = fee
+			}
+			if fee.Cmp(minFee1) <= 0 {
+				minFee1 = fee
+			}
+			totalFee1 = totalFee1.Add(fee)
+			numTxns1++
+		}
+	}
+	avgFee1 := totalFee1.Div64(uint64(numTxns1))
+	if totalFee1.Cmp(expectedFee1) != 0 {
+		t.Error("totalFee1 different than expected fee.", totalFee1.String(), expectedFee1.String())
+		//t.Log(totalFee1.Sub(expectedFee1).HumanString())
+
+	}
+
+	// Mine the next block so we can check the transactions inside
+	block, err = tpt.miner.AddBlock()
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	var totalFee2 types.Currency
+	expectedFee2 := coinFrac.Mul64(13860)
+
+	// Add up total fees
+	numTxns2 := 0
+	maxFee2 := types.SiacoinPrecision.Div64(1000000)
+	minFee2 := types.SiacoinPrecision.Mul64(1000000)
+	for _, txn := range block.Transactions {
+		for _, fee := range txn.MinerFees {
+			if fee.Cmp(maxFee2) >= 0 {
+				maxFee2 = fee
+			}
+			if fee.Cmp(minFee2) <= 0 {
+				minFee2 = fee
+			}
+			totalFee2 = totalFee2.Add(fee)
+			numTxns2++
+		}
+	}
+	avgFee2 := totalFee2.Div64(uint64(numTxns2))
+	if totalFee2.Cmp(expectedFee2) != 0 {
+		t.Error("totalFee2 different than expected fee.", totalFee2.String(), expectedFee2.String())
+		//t.Log(totalFee2.Sub(expectedFee2).HumanString())
+	}
+
+	if avgFee1.Cmp(avgFee2) <= 0 {
+		t.Error("Expected average fee from first block to be greater than average fee from second block.")
+	}
+
+	if totalFee1.Cmp(totalFee2) <= 0 {
+		t.Error("Expected total fee from first block to be greater than total fee from second block.")
+	}
+
+	if numTxns1 < numTxns2 {
+		t.Error("Expected more transactions in the first block than second block.")
+	}
+
+	if maxFee1.Cmp(maxFee2) <= 0 {
+		t.Error("Expected highest fee from first block to be greater than highest fee from second block.")
+	}
+
+	if minFee1.Cmp(maxFee2) < 0 {
+		t.Error("Expected lowest fee from first block to be greater than or equal to than highest fee from second block.")
+	}
+
+	if maxFee1.Cmp(minFee1) <= 0 {
+		t.Error("Expected highest fee from first block to be greater than lowest fee from first block.")
+	}
+
+	if maxFee2.Cmp(minFee2) <= 0 {
```

Comment:
Can drop all of the newlines between these checks, no need to spread them out. Pretty simple actions and it's all the single idea of checking that the state is what it's supposed to be.